### PR TITLE
Bug 1990014: Throw error when debugging Windows nodes

### PR DESCRIPTION
**Pull request by deejross**

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: pkg/cli/debug/debug.go
Comment (aravindhp) [inline pkg/cli/debug/debug.go:972]: Why do you need priviledged containers for debugging a pod? Can't you run `cmd` like you were doing in #907?

**Comment by aravindhp** [comment]
/hold

Adding a hold until we figure out why this is being disabled.

File: pkg/cli/debug/debug.go
Comment (deejross) [reply pkg/cli/debug/debug.go:972]: It doesn't work, and the docs explained that debug needs privileged containers, and that Windows nodes don't currently support that: https://docs.openshift.com/container-platform/4.9/support/troubleshooting/troubleshooting-windows-container-workload-issues.html#accessing-windows-node

This PR is to address a bug where running `oc debug node` currently hangs attempting to debug a Windows node. If we can support debugging Windows nodes, that would be great though.

Comment (aravindhp) [reply pkg/cli/debug/debug.go:972]: That is correct. `oc debug node` does not work for Windows nodes. The main confusion here is the association with [Bug 1990014  - oc debug <pod-name> does not work for Windows pods](https://bugzilla.redhat.com/show_bug.cgi?id=1990014) which has to be debugging Windows Pods. I want to ensure that we are not disabling that.

Comment (deejross) [reply pkg/cli/debug/debug.go:972]: Correct, in that BZ, someone tried both nodes and pods. The intent is to provide users with better feedback when `oc debug node` is attempted on Windows nodes. Once this is merged, I plan to set the bug back to NEW to address the issue with pods, as that is a valid bug.

Comment (selansen) [reply pkg/cli/debug/debug.go:972]: @deejross, Thank you for the detailed explanation.

**Comment by aravindhp** [comment]
/hold cancel

Confirmed with @deejross on Slack that this will not affect pod debugging

**Comment by deejross** [comment]
As a side note, I confirmed that you can debug Windows pods in oc 4.9.15. I was using 4.8.2 and realized #907 hadn't been included in that version.

**Comment by aravindhp** [comment]
> As a side note, I confirmed that you can debug Windows pods in oc 4.9.15. I was using 4.8.2 and realized #907 hadn't been included in that version.

You can `/cherry-pick release-4.8` and get #907 backported given it is a bug fix. It will get picked up in the next 4.8.z.

**Pull request merged**
